src/ussd_optout_rapidpro.js

```diff
@@ -1,29 +1,333 @@
+                            "you opt out. Do you want to delete your info after you stop getting messages?"),
+                error: $(
+                    "Sorry, please reply with the number next to your answer. " +
+                    "Do you want to delete your info after you stop getting messages?"
+                ),
+                choices: [
+                    new Choice("yes", $("Yes")),
+                    new Choice("no", $("No"))
+                ],
+                next: function(choice) {
+                    if(choice.value === "yes") {
+                        return "state_info_deleted";
+                    } else {
+                        return self.states.create("state_trigger_rapidpro_flow");
+                    }
+                }
+            });
+        });
+
+        self.add("state_info_deleted", function(name) {
+            return new ChoiceState(name, {
+                question: $("All your info will be permanently deleted in the next 7 days. " +
+                            "We'll stop sending messages. Please select Next to continue:"),
+                error: $(
+                    "Sorry, please reply with the number next to your answer. " +
+                    "Please select Next to continue:"
+                ),
+                choices: [
+                    new Choice("yes", $("Next"))
+                ],
+                next: "state_trigger_rapidpro_flow"
+            });
+        });
+
+        self.states.add("state_check_previous_optout", function(name) {
+            // Skip this state if they haven't opted out
+            //Skip this state if they have an opted_out reason
+            var contact = self.im.user.get_answer("contact");
+            var optout_reason = contact.fields.optout_reason;
+            if (!self.contact_in_group(self.im.user.get_answer("contact"), self.im.config.optout_group_ids)) {
+                return self.states.create("state_no_previous_optout");
+            }
+            
+            else if((self.contact_in_group(self.im.user.get_answer("contact"), self.im.config.optout_group_ids)) && optout_reason === "") {
+                var question = "Welcome MomConnect. You've opted out of receiving messages from us. Please tell us why:";
+                return new PaginatedChoiceState(name, {
+                    question: $(question),
+                    error: $(question),
+                    accept_labels: true,
+                    options_per_page: null,
+                    characters_per_page: 160,
+                    choices: [
+                        new Choice("miscarriage", $("Miscarriage")),
+                        new Choice("stillborn", $("Baby was stillborn")),
+                        new Choice("babyloss", $("Baby passed away")),
+                        new Choice("no", $("Msgs aren't helpful")),
+                        new Choice("no", $("Other")),
+                        new Choice("no", $("I prefer not to say"))
```

Comment:
We probably want these 3 to still have a value other than `no`, so that we know what the reason is, and we can send this reason to RapidPro when we trigger the flow.

Reply:
okay, will do